### DISCUSSION TITLE
First builds

### DISCUSSION THREAD

**Discussion by Kipio37** [picocomputer/community#20] [Q&A]
Since this is a new project, I thought it would be nice to show off new builds, even though they may look more or less the same.  Here is mine.  I had JLBPCB make the boards, and I sent the BOM to Mouser as suggested on the webpage and ordered everything from there except the Pico's, which they were out of stock of.  So I ordered those from Chicago Electronic Distributors.  Surprisingly I had everything in hand and was able to build the whole thing within one week of ordering everything.  Very easy!  I ordered 5 boards-- not sure what I'm going to do with the rest.

If this discussion isn't appropriate for this category (or these forums in general), don't hesitate to delete it.

![PXL_20230729_215118654](https://github.com/picocomputer/discussions/assets/141074066/1632a14f-3004-4a59-94d8-a14010b4050a)

**Comment by totallymaxed** [comment]
Here is my Founders Edition board built & working that @rumbledethumps very kindly shipped all the way to the UK. The only step that was a little tricky was mounting the VGA socket - there is some misalignment between the mounting pins and the grid of VGA pins. This was easily fixed by adjusting the mounting pins a little. 

The board is working perfectly with no issues post build at all - can wait for the VGA changes that are in the pipeline and to start playing with the board.
![IMG_5053](https://github.com/picocomputer/discussions/assets/4604191/5decfaee-a3e2-4e7e-aafa-40732acf1d54)

**Reply by Technikfreak2002** [reply]
Can i spot a ColorMaximite in the background? :D Have one too!

**Reply by totallymaxed** [reply]
Yes you can 👏

**Comment by brentward** [comment]
I ordered my parts and the board last Saturday. All the parts and the board were here by Thurday. I soldered all the parts on Thursday evening without any issues. On Friday morning I plugged in all the active components and started playing around with it. I got 5 boards from JLCPCB, the parts from Mouser except the Picos, and I got the Picos from Amazon.
![IMG_8259](https://github.com/picocomputer/discussions/assets/22037467/d6af3f09-07bb-4fdc-9b85-c4df8fba1a95)

**Comment by michaelsarr** [comment]
![71306218330__06A45AE0-5189-4CE3-A49F-432C450F54BC](https://github.com/picocomputer/discussions/assets/73266612/24a3094b-541b-4e67-9d2a-7dac2fa64dfc)


![71306217137__6F9CC1B6-08E9-456E-BF2C-27EFF14467BE](https://github.com/picocomputer/discussions/assets/73266612/c8eea22a-851b-4a6a-b4d6-8972ee17702e)


tried to get some code up to load an image but my conversion to 16 colors leaves a lot to be desired (it ia picture of a couple baby possums)
![IMG_6095](https://github.com/picocomputer/discussions/assets/73266612/ecd4d74e-19cb-467c-a695-e41234fdb31f)

i got the board yesterday and soldered it up, had some time to work with it today and get vscode working

boards from JLCPCB , parts from mouser, picos came from a blister pack of them I got when they first came out.. i still had a few left

**Reply by RayGardiner** [reply]
My congratulations and appreciation to Rumbledethumps for a nicely designed 6502 platform and clever integration with the pi Pico. 

I got the boards from JLCPCB,  and ordered the parts from Mouser using the BOM supplied.  Assembly was easy and fast,  the system booted right away and works fine.  I use VSCodium rather than VSCode,  but everything works the same.  I've only compiled the Hello World so far, but that's enough to let me get started.    

![IMG_5325](https://github.com/picocomputer/discussions/assets/2309414/a63ed405-5605-47b0-99c6-d991abe17ada)
![IMG_5326](https://github.com/picocomputer/discussions/assets/2309414/62345209-63af-4cd5-be7f-a68aa25f25ac)

**Reply by brentward** [reply]
I'd like to see the code you used to draw the baby opossums. Would you mind pushing it to your repo?

**Reply by michaelsarr** [reply]
![possum_320x180](https://github.com/picocomputer/discussions/assets/73266612/807ad5f8-6d44-4f24-b0f5-8400ac33b415)
![IMG_6102](https://github.com/picocomputer/discussions/assets/73266612/a5e6da1b-54fb-4dac-92ec-af8f0c2eb6c8)
i made some progress but i dont know how to read from a file ;) 
getting a file not found error so I'm missing something.. want to open a file I  put on the USB thumb

i put some files here if you want to take a look. I use a program running on my laptop to generate a file I include of statements to push the bytes over. but I cant get it to compile with more than half the image.

http://msarr.xyz/~msarr/image_demo/

i would think i could just write it to a binary file then open and read it from the picocomputer . we dont even want to talk about my color conversion logic but im open to suggestions

**Reply by rumbledethumps** [reply]
A 320x240 4bpp image requires 38400 bytes. The C code "RIA_RW0 = 0x00;" turns into something like "LDA #$00; STA RIA_RW0". You'll run out of program space somewhere around 13000 image bytes (13000*5=65000), or about 320x80 pixels. That's why the compiler refuses to do the whole image.

**Reply by michaelsarr** [reply]
thats what I was thinking, thanks for the input. ill have to look into trying to open the file and read it in or do something different. It was really just something to try.

**Comment by Technikfreak2002** [comment]
I just finished my build! :D So far everything works fine! Board form JLCPCB and parts from Mouser some came out of my "old parts" box :D
Maybe this is the first PicoComputer in Germany? 

![WhatsApp Image 2023-08-16 at 19 05 58](https://github.com/picocomputer/discussions/assets/38852340/3d44e311-c8ae-4e5a-be35-9e59ab14334e)

**Reply by rumbledethumps** [reply]
The Samsung KM681000ALP is a 5V part. Also, you have the 100ns version. And you're missing a bunch of 0.1uF caps. Does this work?

**Reply by Technikfreak2002** [reply]
Yes this works fine even at 8Mhz! 

I did not order the caps. I thought i had enough... now i have to reorder.

The RAM was from an old printer. I only want to test if it works.
I already replaced it with the 3.3V version you use in your schematic.

**Reply by rumbledethumps** [reply]
I'd like to take credit for a robust design that works with any random parts from your junk bin, but Samsung was probably building these chips on a fab that was 20 years newer than the datasheet.

**Reply by Technikfreak2002** [reply]
I made all my first tests with this old RAM. First on a breadboard with 4 MHz max. I had no AC logic chips laying around. But as soon as the PCB arrived i used this RAM. Full speed. Test software was ehBasic. No issues. 
This is by far not in the save operation spec's... but for first tests it's fine! :D

This is not the first RAM that surprised me. I had to deal with an old NVRAM at work. We made a backup of the old RAM and replaced it with a 'new' one. After that we took it apart and mesured the battery voltage with 1.23V.

Yes your design is very robust! You showed it all in your videos! Up to 8MHz on a breadboard is insane!

**Comment by Nordlaw-tech** [comment]
I've built two - one for my son in law - a professional software developer.
![PXL_20230822_151553651](https://github.com/picocomputer/discussions/assets/66093324/078cae2b-d956-46a1-9b7d-35606c21e829)

**Comment by michaelsarr** [comment]
![IMG_6155](https://github.com/picocomputer/discussions/assets/73266612/2e1df342-8530-4715-b438-96ce90de7dd7)
Since I had to order 5 pcb I put together another

**Comment by pwillard** [comment]
It worked on the first power-up... so an excellent Project!

![20230923_130500](https://github.com/picocomputer/discussions/assets/3194144/48d3e8d9-f529-45f6-922e-9ae8a98206b4)

**Comment by vruumllc** [comment]
Got my pcboard from Tindie and parts from Mouser. 
Built it, and it worked the first time at 8 MHz.
Very nice design, and the modern workflow is great!

The only thing I hate is all the dangling bits, so I am going to design a 3-D printed brick-like case,
with integrated hub and flip-up screen.

**Reply by vruumllc** [reply]
Step 1 complete. Designed and 3D printed a simple case to unite rp6502, usb hub, and 12vdc to 5vdc converter.
Next, I plan to incorporate 5in. display shown, stereo speakers, and an amp.

![rp6502_case1_tonyvr](https://github.com/picocomputer/discussions/assets/145504135/461d893a-035d-4bf4-8d8a-4736ba80be39)

![rp6502_case2_tonyvr](https://github.com/picocomputer/discussions/assets/145504135/6faa1ff9-c96d-4479-a19b-0fb38a510642)

**Reply by vruumllc** [reply]
OK, step 2 complete: Integrated, video, audio, and hub:
![tonyvr_front](https://github.com/picocomputer/discussions/assets/145504135/661b3e88-0d2b-4246-84ea-5884f5d47572)
![tonyvr_back](https://github.com/picocomputer/discussions/assets/145504135/c9bb1d7f-791e-4c3a-a5e5-49f2a2ddfa94)
![tonyvr_inside](https://github.com/picocomputer/discussions/assets/145504135/26b1df7f-ce23-4e7c-8995-3f9aba4d9c99)
![rp6502_hub_video_audio_inside_display_photo](https://github.com/picocomputer/discussions/assets/145504135/3aafe5a8-91f9-4f65-9f7a-dcfb12591471)

Next, I'll be trying to add a rechargeable battery.

**Reply by rumbledethumps** [reply]
Awesome. Very tidy! SO much detail in the 3D models.

**Reply by vruumllc** [reply]
Thanks! That means a lot coming from a craftsman such as yourself.

Now that this is done I am going to use it as the basis for 2 more cases.

1. Case for just a RP6502 board with option for solid or see-through cover.
2. Case for RP6502 plus USB hub I used, with option for solid or see-through cover.

BTW, I designed it so you can reuse the plexiglass case you ship for the see-through cover.

Once I get this done I'll post designs, STL files, and bill of materials for all the cases.

**Reply by vruumllc** [reply]
OH, a question! Does the RP6502 have an 'official' color / logo? 

I need to order more Tough PLA filament anyway. 

(The glow-in-the-dark green was all I had enough of to print this project. This color, though, is the NVidia Jetson color.)

**Reply by rumbledethumps** [reply]
It has a spirit animal: Didelphis virginiana

**Reply by vruumllc** [reply]
OK, I looked that up. Pretty funny.

**Reply by vruumllc** [reply]
Step 3: Complete!  I finally got battery and charger.  

Look Ma! No Wires!

![rp6502_portable_front](https://github.com/picocomputer/community/assets/145504135/4c8630a1-a882-486a-ac32-722eb62f2f97)

![rp6502_portable_rear](https://github.com/picocomputer/community/assets/145504135/521e6ae8-8f73-4386-afd8-692f00b1ee36)
 
Battery life is about 12 hours.

**Comment by pwillard** [comment]
Nice work!

**Reply by vruumllc** [reply]
Thanks!

**Comment by ronaldlens** [comment]
Extremely stoked!
Just received my board from PCBWay. I went the lazy route and had one soldered :). Just waiting for Mouser to deliver their part and I should be in business. I do have 4 extra PCBs here in the Netherlands, so if anyone within the area would like one, please contact me.

Thanks @rumbledethumps for such an elegant design! BTW can fully recommend PCBWay, the assembly took a bit longer but all the files by @rumbledethumps worked like a charm.The whole order came to about USD125 including shipping via DHL so pretty reasonable imho.

![IMG_7475](https://github.com/picocomputer/discussions/assets/2178191/e1f78af5-bf37-41e2-9442-988abf16e349)

**Comment by WojciechGw** [comment]
[@rumbledethumps](https://github.com/rumbledethumps) thank You very much for this excellent and exciting project :)

first build
![1698948233001](https://github.com/picocomputer/discussions/assets/7420761/d8831e98-b410-44e4-a411-972b35753f7a)

parts : JLCPCB, Mouser

**Comment by Hyenadae** [comment]
Greetings from Bermuda, and now another kit owner! 

I just slotted the chips into my PicoComputer and re-used the VIA and 6502 from my Ben Eater kit (remains of the build on the breadboard, lol) and it seems to be working fine. The case is also on it now, and I got a USB hub to let me use a flash drive with the kinda lame folding keyboard.
![HyPicoComputer6502-Dec2-23](https://github.com/picocomputer/community/assets/3067205/691a8135-1d9e-42ce-a5ff-5eb2fcc846a5)

Need to get the dev env on my PC set up, and try out some of the demos as I was a bit worried about the ICs being properly in the socket, but so far it seems operational

**Reply by vruumllc** [reply]
Welcome! I have been having so much fun with this. I hope you do too! Check out my tetris game. There is a built executable in the zip file, so you can try it out without a working dev environment.

**Reply by Hyenadae** [reply]
Thanks! I'm kind of low on spare screens, do you know of any programs or games that can run over the USB serial link in text only mode?
Kinda ironic with a VGA orientated system. I think I'll get BASIC running, and see if I can run some fun calculations as a 'stress test'

**Reply by vruumllc** [reply]
Here is adventure, and old text game:
[adventure.zip](https://github.com/picocomputer/community/files/13537738/adventure.zip)

**Reply by vruumllc** [reply]
Here's basic, while I'm at it.
[basic.zip](https://github.com/picocomputer/community/files/13537752/basic.zip)

Try:
10 i =0
20 PRINT i
30 i = i + 1
40 GOTO 20

That is about the extent of my basic. Enjoy!

**Reply by netzerohero** [reply]
Hi and welcome!  In addition to @vruumllc's cool Tetris game, and the Adventure game he pointed out, here's a historical but more meaty basic program I dusted-off (Eliza) and have been tinkering with:

[ELIZA01c.BAS.zip](https://github.com/picocomputer/community/files/13537883/ELIZA01c.BAS.zip)

ELIZA is text-only, but runs both on the VGA-console as well as from usb/serial/terminal.  (Better in color on VGA though.)

EhBASIC remains a work-in-progress; you'll have to load basic source via your favorite terminal program.  You'll likely have to tune newline delays for uploading very large basic source like I attached.  (I use 75mSecs on linux/minicom.)

Hope this helps, enjoy!

**Reply by Hyenadae** [reply]
I'm making progress, the USB is working (had an issue earlier), and I used the python upload script to load it to my flash drive
Here's my 6502 at 8Mhz calculating 100 digits of Pi in EHBasic :)

Now to run Eliza01c
![image](https://github.com/picocomputer/community/assets/3067205/3ad4b1e6-bee5-4c39-87ac-ccf0b8ff5beb)

There seems to be an issue with the Ehbasic SQR function I assume it may not be implemented yet, as it's giving me an error (Babbage Problem code) if I try to run AppleSoft basic (that syntax worked for Pi) from here: https://rosettacode.org/wiki/Category:Applesoft_BASIC

**Reply by netzerohero** [reply]
Cool rosettacode website; thanks for sharing!

Here's a link to EhBASIC's User Manual:
http://retro.hansotten.nl/uploads/leedavison/Enhanced_6502_BASIC_reference_manual.pdf

Manual mentions SQR(); appears to be implemented, see assembly source file lang.s, label: LAB_SQR:
https://github.com/picocomputer/ehbasic/blob/main/src/lang.s

Suggestion: spin-off a separate thread for EhBASIC discussion. It would be a good thread to start in gathering EhBASIC specifics.

**Comment by n9jcv** [comment]
My first build, not in a case yet, that is another project LOL.

![17040492383618782198240923248539](https://github.com/picocomputer/community/assets/6472203/d87820af-9bbf-4f5d-a12d-d45a69c2955d)

Thanks to @rumbledethumps for sharing the great design and thanks to the community for the help!

**Comment by Zulu0001** [comment]
Today was the day.  I finally found some time to build the machine.  It even worked first try.   Just happened the first keyboard I pulled out of my pile worked with a micro usb to A OTG adapter (Amazon basics ps2 compatible usb).  I tried a Dell keyboard, it didn't work.  My hub didn't work either.  If anyone has had success let me know what it is.  I would like to pick one up on Amazon.  Can't wait to play...

![Picocomputer 1](https://github.com/picocomputer/community/assets/8394646/4fc2958e-e4ef-4da9-8131-509476fc7a8f)
![Picocomputer 6](https://github.com/picocomputer/community/assets/8394646/8f9c1766-0961-436d-9fd7-1c84c6abb9bc)
![Picocomputer 2](https://github.com/picocomputer/community/assets/8394646/cec68476-0590-4b36-97f9-541724e04426)
![Picocomputer 8](https://github.com/picocomputer/community/assets/8394646/743b56ad-590c-41bb-94ee-1f6c70983876)

**Reply by vruumllc** [reply]
This wireless keyboard and mouse work well:

https://www.amazon.com/Macally-Wireless-Keyboard-Mouse-Combo/dp/B081549BN2/ref=sr_1_3?crid=1XPV3IBH3QEHS&keywords=Macally%2BSmall%2BWireless%2BKeyboard%2Band%2BMouse%2BCombo%2Bfor%2BPC&qid=1707279745&s=electronics&sprefix=macally%2Bsmall%2Bwireless%2Bkeyboard%2Band%2Bmouse%2Bcombo%2Bfor%2Bpc%2Celectronics%2C162&sr=1-3&th=1

I use this nice little hub in my cases:

Vilros Micro USB to USB 4 Port OTG Hub, https://www.ebay.com/itm/313146204417, [$7]

**Reply by Zulu0001** [reply]
Thanks vruumllc, that will save me a bunch of time :-)

**Reply by netzerohero** [reply]
This hub works for me: 
AuviPal 3-Port Micro USB OTG Hub Adapter, [(https://www.amazon.com/gp/product/B083WML1XB/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1)](https://www.amazon.com/gp/product/B083WML1XB/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1), [$8].  Also available on ebay.

**Reply by Zulu0001** [reply]
Thanks netzerohero good to know there are options.  I had this one in my Amazon wish list.

**Comment by n9jcv** [comment]
I had some difficulties with hubs and keyboards initially.  I think I tried 3 of each before finding a combo that works.  I did not buy any, just tried all I had in the basement and finally had success.

**Comment by Zulu0001** [comment]
New home while it's not being used
![Picocomputer 9](https://github.com/picocomputer/community/assets/8394646/61ea1bb8-c25c-4783-a628-52b275d62bcf)

**Comment by basic-xl** [comment]
Hello everyone. I completed my build some time back, and just never noticed this forum. My bad, lol. I ordered my pcb's from JLCPCB, and the parts from Mouser, except for the Pi's which I had lying around from another project. I also was successful on my first build, and was astonished (at myself and my skillset) it powered up properly the first time. I truly wish I knew more about all this, as I gather it's great fun. I know enough 6502 Assembly to get myself into trouble 🙄 but not enough to be actually useful. Perhaps I'll look at some of the other projects


![20240303_085246](https://github.com/picocomputer/community/assets/112533139/1ba81b44-0419-4a6f-9324-4cda2ba31cf4)

As you can see, I hung mine on the wall beside my monitor. My nephew runs a phone repair / gaming business and 3d design/printing is one of his forte's (when he's not too busy lol) so he designed and printed the mounting hardware. Thanks Mikah!

![20240302_091546](https://github.com/picocomputer/community/assets/112533139/fbf342f4-35cf-43d8-88ce-5529808deb0c)

**Comment by Bee91303** [comment]
I just stuffed 2 out of 5 I had made by PCBway, working great!

![IMG_5333](https://github.com/picocomputer/community/assets/33590462/21d8c158-d469-469b-8820-66c4d85c803e)
